apps/aechannel/src/aesc_fsm.erl

```diff
@@ -206,16 +211,19 @@ inband_msg(Fsm, To, Msg) ->
 initiate(Host, Port, #{} = Opts0) ->
     lager:debug("initiate(~p, ~p, ~p)", [Host, Port, aesc_utils:censor_init_opts(Opts0)]),
     Opts = maps:merge(#{client => self(), role   => initiator}, Opts0),
-    try init_checks(Opts) of
-        ok ->
-            aesc_fsm_sup:start_child([#{ host => Host
-                                       , port => Port
-                                       , opts => Opts }]);
-        {error, _Reason} = Err ->
-            Err
-    ?CATCH_LOG(_E)
-        {error, _E}
-    end.
+    Proceed = fun() ->
+                      try init_checks(Opts) of
+                          ok ->
+                              aesc_fsm_sup:start_child([#{ host => Host
+                                                         , port => Port
+                                                         , opts => Opts }]);
+                          {error, _Reason} = Err ->
+                              Err
+                      ?CATCH_LOG(_E)
+                          {error, _E}
+                      end
+              end,
+    check_limits(Proceed, Opts).
```

Comment:
Why is the `check_limits` check not added to `init_checks` instead?

Reply:
Partly because `init_checks()` is checking the static properties of the provided arguments, while the `check_limits()` function checks the dynamic headroom of the node, and partly because the dynamic check should be performed as late as possible - or as early as possible. There are valid arguments for both, where an argument for early checking would be that the node should minimize the amount of work spent on rejected requests, and an argument for late checking might be that we want to treat new channels and 'returning' channels differently. Either way, the limit check is different from the parameter checking.

Reply:
I agree that the limit check should be performed early. However, the same result could be achieved by simply appending it to the list of init checks, since `aeu_validation:run/1` will stop as soon as a check fails.